src/common/com/intellij/plugins/haxe/ide/HaxeCompilerCompletionContributor.java

```diff
@@ -120,16 +123,24 @@ protected void addCompletions(@NotNull CompletionParameters parameters,
                        //Export/flash/haxe contains build.hxml which gets generated after build
                        break;
                      case HaxeModuleSettingsBaseImpl.USE_OPENFL:
-                       commandLineArguments.add(HaxelibCommandUtils.getHaxelibPath(moduleForFile));
-                       commandLineArguments.add("run");
-                       commandLineArguments.add("lime");
-                       commandLineArguments.add("display");
-                       //flash, html5, linux, etc
                        String targetFlag = moduleSettings.getNmeTarget().getTargetFlag();
-                       commandLineArguments.add(targetFlag);
 
-                       List<String> stdout = HaxelibCommandUtils.getProcessStdout(commandLineArguments,
-                                                                                  BuildProperties.getProjectBaseDir(project));
+                       List<String> stdout = openFLDisplayArguments.get(moduleForFile.getModuleFilePath() + targetFlag);
+                       if (stdout == null) {
```

Comment:
The test is going to have to be a lot more robust than "stdout==null".  As it is, any changes in the entire module will be unavailable for completion until IDEA is restarted.  Often, developers expect completion from the variables they just declared.  This solution makes it so the compiler can't help.  I didn't look to see if the 'old' haxe completion engine is still called, or if it was just replaced when you added the compiler completion.  If it was the first case, then we might be able to get away with this for the short term.

I don't know enough about the compiler's completion yet.  Doesn't it only give an answer for the code position you first asked for?  If that's the case, and with this check, we'll always see the answer from the first request, and almost never get the answer for the current cursor position.  Or am I missing something?